src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -123,6 +124,29 @@ sol_str_slice_to_string(const struct sol_str_slice slice)
     return strndup(slice.data, slice.len);
 }
 
+static inline void
+sol_str_slice_remove_leading_whitespace(struct sol_str_slice *slice)
```

Comment:
I'd rather have a funciton that takes a `struct sol_str_slice` (non-pointer) and returns another `struct sol_str_slice` (non-pointer).